tool/gh_actions/install_synthesis.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# install_synthesis.sh
+# GitHub Actions step: Install project dependencies.
+#
+# 2023 May 09
+# Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+#
+
+set -euo pipefail
+
+sudo apt-get update
+DEBIAN_FRONTEND=noninteractive sudo apt-get install -y \
+	       build-essential \
+	       clang \
+	       bison \
+	       flex \
+	       libreadline-dev \
+	       gawk \
+	       tcl-dev \
+	       libffi-dev \
+	       git \
+	       pkg-config \
+	       python3 \
+	       python3-dev \
+	       python3-pip \
+	       python3-setuptools \
+	       python3-wheel \
+	       python3-tk
+
+sudo git clone https://github.com/YosysHQ/yosys.git
+cd yosys
+sudo make -j$(nproc)
```

Comment:
Perhaps we can use one of the pre-built yosys binaries to speed this up:
https://yosyshq.net/yosys/download.html

Reply:
This is now done, I pull down the OSS CAD suite.